### DISCUSSION TITLE
Sorting collection by multiple columns

### DISCUSSION THREAD

**Discussion by Gelidan** [yajra/laravel-datatables#2547] [Q&A] [answered]
### Summary of problem or feature request

Hello, 
First of all, thanks for the package, I realy enjoy using it!

I have the following problem:
I created a DataTable class, for my users, and use the builder, to generate everything, and it's working almost perfectly, except the ordering.
I have my data in collection, not in elloquent. So I can't use the OrderColumns api. 
I want to order the results by multiple columns, but the builder's ->orderBy() method seems to be buggy.
In the HasOptions trait, you check, if the index is array, but don't iterate over it. And I'm not able to get it to work. 
I wanted to order by the 8th and 1st column, both times ascending. 

### Code snippet of problem

From the DataTable class I use the builder

```php
public function html() {
        return $this->builder()
            ->setTableId('users-table')
            ->addTableClass('vexor-dt-table')
            ->columns($this->getColumns())
            ->minifiedAjax()
            ->dom('Bfrtip')
            ->language(['url' => '//cdn.datatables.net/plug-ins/9dcbecd42ad/i18n/Hungarian.json'])
            ->responsive(true)
            ->autoWidth(false)
            ->orderBy([8,1], 'asc')
            ->columnDefs([
                ['responsivePriority' => 1, 'targets' => -1],
            ])
            ->rowGroupDataSrc('department_id')
            ->lengthMenu([25, 50, 100, 200])
            ->buttons(
                Button::make(['extend' => 'create', 'text' => '<i class="fas fa-plus"></i> ' . trans('global.app_add_new'), 'className' => 'btn-outline-success']),
                Button::make(['extend' => 'excel', 'text' => '<i class="fas fa-file-excel"></i> ' . trans('global.app_excel'), 'className' => 'btn-outline-info']),
                Button::make(['extend' => 'pdfHtml5', 'orientation' => 'landscape', 'text' => '<i class="fas fa-file-pdf"></i> ' . trans('global.app_pdf'), 'className' => 'btn-outline-info']),
                Button::make(['extend' => 'print', 'text' => '<i class="fas fa-print"></i> ' . trans('global.app_print'), 'className' => 'btn-outline-info']),
                Button::make(['extend' => 'reload', 'text' => '<i class="fas fa-sync-alt"></i> ' . trans('global.app_reload'), 'className' => 'btn-outline-info']),
                Button::make(['extend' => 'pageLength', 'text' => '<i class="fas fa-list"></i> ' . trans('cafeteria_settings.words.home.rows'), 'className' => 'btn-outline-info']),
            );
    }
```

And in the Trait, your code is this (HasOptions.php orderBy function)

```php
public function orderBy($index, $direction = 'desc')
    {
        if ($direction != 'desc') {
            $direction = 'asc';
        }

        if (is_array($index)) {
            $this->attributes['order'][] = $index;
        } else {
            $this->attributes['order'][] = [$index, $direction];
        }

        return $this;
    }
```

I managed to "fix" the problem with this foreach

```php
public function orderBy($index, $direction = 'desc')
    {
        if ($direction != 'desc') {
            $direction = 'asc';
        }

        if (is_array($index)) {
            foreach ($index as $values){
                $this->attributes['order'][] = [$values, $direction];
            }
        } else {
            $this->attributes['order'][] = [$index, $direction];
        }

        return $this;
    }
```
But it's a bad fix, for overwrite something in vendor. 
How can I overwrite this Trait function from controller?

Thanks, for the help!

### System details

- Operating System: Windows 10 with XAMPP
- PHP Version: 7.4
- Laravel Version: 7.0
- Laravel-Datatables Version: 1.5

**Comment by yajra** [comment, marked as answer]
Use `order` instead of `orderBy`

```php
->order([[8, 'asc'], [1, 'asc']])
```

**Reply by Gelidan** [reply]
Thanks for the help! It's working. Only one small correction:

The Builder does not have an order function, it has an orders function

`->orders([[8, 'asc'], [1, 'asc']])`

Thanks once more, and keep up maintaining this awesome package! Happy Holidays!

**Reply by yajra** [reply]
Also, I think you can chain orderBy to have multiple sort.

```php
->orderBy(8, 'asc')
->orderBy(1, 'asc')
```

Thanks and Happy Holidays too!

**Reply by Gelidan** [reply]
I would swear I tried this, and didn't work, but now it's working. Maybe I was too crumped up.
Thanks
